### PR TITLE
Fix: don't override user definied `region`

### PR DESCRIPTION
**Pull request by zzxwill**
Fix #https://github.com/oam-dev/kubevela/issues/3384

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Comment by anoop2811** [comment]
Any idea when this will be merged?

File: pkg/appfile/appfile.go [hunk 1]
```diff
@@ -669,7 +671,7 @@ func generateTerraformConfigurationWorkload(wl *Workload, ns string) (*unstructu
 	}
 
 	configuration := terraformapi.Configuration{
-		TypeMeta: metav1.TypeMeta{APIVersion: "terraform.core.oam.dev/v1beta1", Kind: "Configuration"},
+		TypeMeta: metav1.TypeMeta{APIVersion: "terraform.core.oam.dev/v1beta2", Kind: "Configuration"},
```
Comment (wonderflow) [inline pkg/appfile/appfile.go:674]: do we need to do some migrate here?

File: pkg/appfile/appfile.go [hunk 2]
```diff
@@ -682,8 +684,6 @@ func generateTerraformConfigurationWorkload(wl *Workload, ns string) (*unstructu
 	switch wl.FullTemplate.Terraform.Type {
 	case "hcl":
 		configuration.Spec.HCL = wl.FullTemplate.Terraform.Configuration
-	case "json":
-		configuration.Spec.JSON = wl.FullTemplate.Terraform.Configuration
```
Comment (wonderflow) [inline pkg/appfile/appfile.go:686]: will it cause incompatible here?

Comment (zzxwill) [reply pkg/appfile/appfile.go:686]: [JSON support was deprecated in v0.3.1, and removed in v0.4.6 in Terraform Controller](https://github.com/oam-dev/terraform-controller#supported-terraform-configuration) for these reasons:
- JSON is not friendly for humans who prepare HCL in Terraform Controller or ComponentDefinition in KubeVela: nearly all Terraform modules are written in HCL.
- No user requirements: No feedback from a single user was received.

File: pkg/appfile/appfile.go (hunk 1 above)
Comment (zzxwill) [reply pkg/appfile/appfile.go:674]: Yes, I did in https://github.com/kubevela/kubevela/pull/3503/files/cde8d3a9577d1049b4325b52427b8ec2c2853653#diff-abd52d53959b7e73bba4c28566a77193d4ac241dc02f1fc17e149dbb8fa4725eL237-R244.

File: pkg/appfile/appfile.go (hunk 2 above)
Comment (wonderflow) [reply pkg/appfile/appfile.go:686]: ok

**Review by wonderflow — APPROVED** [review]
LGTM, please make sure these case have compatibility and provide migration guide.

**Comment by zzxwill** [comment]
By upgrading vela-core (to version `1.3.4`):
All existing Terraform typed Applications won't be affected.
But all newly provisioned Terraform typed Applications won't become successful. To solve the issue:
- Upgrade add-on `terraform` to version `1.0.6+`
- Manually upgrade CRD Configuration https://github.com/oam-dev/terraform-controller/blob/v0.4.3/chart/crds/terraform.core.oam.dev_configurations.yaml.

**Review by leejanee — APPROVED** [review]
LGTM

**Review by Somefive — APPROVED** [review]

**Pull request merged**